pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -334,7 +334,7 @@ void messageAcked(CommandAck ack) {
                 return;
             }
 
-            if (subType == SubType.Shared) {
+            if (subType == SubType.Shared || subType == SubType.Key_Shared) {
```

Comment:
Since this is being checked multiple times, we could have a static method to check for this condition

Reply:
Exactly this is being checked multiple times, i'm already add static method named Subscription.isIndividualAckMode(), PTAL